core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -2871,22 +2866,25 @@ public String[] getUserListFromProperties(String property, String value, String
                             }
                         }
                         String propertyValue = attrBuffer.toString();
-                        Attribute serviceNameObject = attributes.get(serviceNameAttribute);
-                        String serviceNameAttributeValue = null;
-                        if (serviceNameObject != null) {
-                            serviceNameAttributeValue = (String) serviceNameObject.get();
-                        }
                         // Length needs to be more than userAttributeSeparator.length() for a valid
                         // attribute, since we
                         // attach userAttributeSeparator.
-                        if (propertyValue != null && propertyValue.trim().length() > userAttributeSeparator.length()) {
-                            if (LDAPConstants.SERVER_PRINCIPAL_ATTRIBUTE_VALUE.equals(serviceNameAttributeValue)) {
-                                continue;
+                        if (propertyValue == null || propertyValue.trim().length() <= userAttributeSeparator.length()) {
+                            continue;
+                        }
+                        if (shouldSkipServicePrincipals()) {
+                            Attribute serviceNameObject = attributes.get(serviceNameAttribute);
+                            String serviceNameAttributeValue = null;
+                            if (serviceNameObject != null) {
+                                serviceNameAttributeValue = (String) serviceNameObject.get();
```

Comment:
Is there any possibilities for Null value?

Reply:
Check other places as well

Reply:
There is a null check in place.